apps/ui/src/composables/useSharing.ts

```diff
@@ -0,0 +1,122 @@
+import { FunctionalComponent } from 'vue';
+import { Proposal, Space, User } from '@/types';
+import ICFarcaster from '~icons/c/farcaster';
+import ICLens from '~icons/c/lens';
+import ICX from '~icons/c/x';
+
+type SpaceUser = { user: User; space: Space };
+type SocialNetwork = 'x' | 'lens' | 'farcaster';
+
+export type ShareableType = 'proposal' | 'user' | 'space-user';
+export type PayloadType = User | Proposal | SpaceUser;
```

Comment:
```suggestion
export type Payload = User | Proposal | SpaceUser;
```

nit. I think it makes more sense like this, `ShareableType` makes sense to have `*Type` suffix as it **is** type, of shareable, here it's just payload.

Reply:
Renamed to `Payload` and `PayloadType`

Reply:
Feel free to merge once acknowledged

Reply:
`Payload` and `PayloadType` are great.